lib/segment/tests/integration/payload_index_test.rs

```diff
@@ -248,6 +248,81 @@ fn build_test_segments_nested_payload(path_struct: &Path, path_plain: &Path) ->
     (struct_segment, plain_segment)
 }
 
+fn test_search_comparison<F>(mut param_builder: F)
+where
+    F: FnMut(&mut rand::rngs::ThreadRng, usize) -> (Vec<f32>, Filter),
```

Comment:
I like this extraction. Though this way of getting arguments feels like overcomplication to me, why not introduce a static const for dimensions and take regular arguments?

Reply:
done.